File: implementation/scipamato/core/persistence-jooq/src/intTest/java/ch/difty/scipamato/core/persistence/JooqUserRoleRepoIntegrationTest.java
package ch.difty.scipamato.core.persistence;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Set;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jooq.JooqTest;
import org.testcontainers.junit.jupiter.Testcontainers;

import ch.difty.scipamato.core.auth.Role;
import ch.difty.scipamato.core.persistence.user.JooqUserRoleRepo;

@JooqTest
@Testcontainers
class JooqUserRoleRepoIntegrationTest {

    @Autowired
    private JooqUserRoleRepo repo;

    @Test
    void findingRolesForUser_withTwoRoles_returnsBoth() {
        final Set<Role> roles = repo.findRolesForUser(4);
        assertThat(roles).containsExactlyInAnyOrder(Role.ADMIN, Role.USER);
    }

    @Test
    void findingRolesForUser_withNoRoles_returnsNone() {
        final Set<Role> roles = repo.findRolesForUser(7);
        assertThat(roles).isEmpty();
    }
}

File: implementation/scipamato/core/persistence-jooq/src/intTest/kotlin/ch/difty/scipamato/core/persistence/JooqUserRoleRepoIntegrationTest.kt
package ch.difty.scipamato.core.persistence

import ch.difty.scipamato.core.auth.Role
import ch.difty.scipamato.core.persistence.user.JooqUserRoleRepo
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jooq.JooqTest
import org.testcontainers.junit.jupiter.Testcontainers

@Suppress("FunctionName", "MagicNumber")
@JooqTest
@Testcontainers
internal open class JooqUserRoleRepoIntegrationTest {

    @Autowired
    private val repo: JooqUserRoleRepo? = null

    @Test
    fun findingRolesForUser_withTwoRoles_returnsBoth() {
        val roles = repo!!.findRolesForUser(4)
        assertThat(roles).containsExactlyInAnyOrder(Role.ADMIN, Role.USER)
    }

    @Test
    fun findingRolesForUser_withNoRoles_returnsNone() {
        val roles = repo!!.findRolesForUser(7)
        assertThat(roles).isEmpty()
    }
}
